worlds/cv64/__init__.py

```diff
@@ -0,0 +1,313 @@
+            self.branching_stages = True
+
+        self.active_stage_exits = get_normal_stage_exits(self)
+
+        stage_1_blacklist = []
+
+        # Prevent Clock Tower from being Stage 1 if more than 4 S1s are needed to warp out of it.
+        if self.s1s_per_warp > 4 and not self.options.multi_hit_breakables:
+            stage_1_blacklist.append(rname.clock_tower)
+
+        # Shuffle the stages if the option is on.
+        if self.options.stage_shuffle:
+            self.active_stage_exits, self.starting_stage, self.active_stage_list = \
+                shuffle_stages(self, stage_1_blacklist)
+        else:
+            self.starting_stage = rname.forest_of_silence
+            self.active_stage_list = [stage for stage in vanilla_stage_order if stage in self.active_stage_exits]
+
+        # Create a list of warps from the active stage list. They are in a random order by default and will never
+        # include the starting stage.
+        self.active_warp_list = generate_warps(self)
+
+    def create_regions(self) -> None:
+        # Add the Menu Region.
+        reg_names = ["Menu"]
+
+        # Add every stage Region by checking to see if that stage is active.
+        reg_names.extend([name for name in get_region_names(self.active_stage_exits)])
+
+        # Add the Renon's shop Region if shopsanity is on.
+        if self.options.shopsanity:
+            reg_names.append(rname.renon)
+
+        # Add the Dracula's chamber (the end) Region.
+        reg_names.append(rname.ck_drac_chamber)
+
+        # Set up the Regions correctly
+        self.multiworld.regions.extend([Region(name, self.player, self.multiworld) for name in reg_names])
+
+        # Add the warp Entrances to the Menu Region.
+        self.get_region("Menu").add_exits(get_warp_entrances(self.active_warp_list))
+
+        for reg_name in reg_names:
+
+            # Add the Locations to all the Regions.
+            loc_names = get_region_info(reg_name, "locations")
+            if loc_names is not None:
+                verified_locs, events = verify_locations(self.options, loc_names)
+                self.get_region(reg_name).add_locations(verified_locs, CV64Location)
+
+                # Place event Items on all of their associated Locations.
+                for event_loc in events:
+                    self.get_location(event_loc).place_locked_item(self.create_item(events[event_loc], "progression"))
+                    # If we're looking at a boss kill trophy, increment the total S2s and, if we're not already at the
+                    # set number of required bosses, the total required number. This way, we can prevent gen failures
+                    # should the player set more bosses required than there are total.
+                    if events[event_loc] == iname.trophy:
+                        self.total_s2s += 1
+                        if self.required_s2s < self.options.bosses_required.value:
+                            self.required_s2s += 1
```

Comment:
```suggestion
                for event_loc, event_item in events.items():
                    self.get_location(event_loc).place_locked_item(self.create_item(event_item, "progression"))
                    # If we're looking at a boss kill trophy, increment the total S2s and, if we're not already at the
                    # set number of required bosses, the total required number. This way, we can prevent gen failures
                    # should the player set more bosses required than there are total.
                    if event_item == iname.trophy:
                        self.total_s2s += 1
                        if self.required_s2s < self.options.bosses_required.value:
                            self.required_s2s += 1
```

Reply:
alternatively something like https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/messenger/subclasses.py#L36

Reply:
I will probably go with your first suggestion, as importing CV64Item into locations.py would cause a circular import with the way it's set up now (items.py imports base_id and get_location_info from locations.py).